src/hooks/usePatchStatusSelect.ts

```diff
@@ -113,72 +127,90 @@ export const usePatchStatusSelect = (
-      } else {
-        // fallback in case web workers are not available.
-        // This code reflects logic in public/web_worker/patchBuildVariantReduce.js
-        // Iterate through PatchBuildVariants and determine if a task should be
-        // selected or not based on if the task status correlates with the 2 filters.
-        // if 1 of the 2 filters is empty, ignore the empty filter
-        const baseStatuses = new Set(baseStatusFilterTerm);
-        const statuses = new Set(patchStatusFilterTerm);
-        const nextState =
-          patchBuildVariants?.reduce(
+      // Iterate through PatchBuildVariants and determine if a task should be
+      // selected or not based on if the task status correlates with the 2 filters.
+      // if 1 of the 2 filters is empty, ignore the empty filter
+      let baseStatuses = new Set(baseStatusFilterTerm[versionId]);
+      let statuses = new Set(patchStatusFilterTerm[versionId]);
+
+      let nextState =
+        patchBuildVariants?.reduce(
+          (accumA, patchBuildVariant) =>
+            patchBuildVariant.tasks?.reduce(
+              (accumB, task) => ({
+                ...accumB,
+                [task.id]:
+                  (!!baseStatusFilterTerm[versionId]?.length ||
+                    !!patchStatusFilterTerm[versionId]?.length) &&
+                  (patchStatusFilterTerm[versionId]?.length
+                    ? statuses.has(task.status)
+                    : true) &&
+                  (baseStatusFilterTerm[versionId]?.length
+                    ? baseStatuses.has(task.baseStatus)
+                    : true),
+              }),
+              accumA
+            ),
+          { ...selectedTasks[versionId] }
+        ) ?? {};
+
+      const newTaskSelect = { [versionId]: nextState };
+
+      childVersions?.forEach((cv) => {
+        baseStatuses = new Set(baseStatusFilterTerm[cv.id]);
+        statuses = new Set(patchStatusFilterTerm[cv.id]);
+        nextState =
+          cv.buildVariants?.reduce(
             (accumA, patchBuildVariant) =>
               patchBuildVariant.tasks?.reduce(
                 (accumB, task) => ({
                   ...accumB,
                   [task.id]:
-                    (!!patchStatusFilterTerm?.length ||
-                      !!baseStatusFilterTerm?.length) &&
-                    (patchStatusFilterTerm?.length
+                    (!!patchStatusFilterTerm[cv.id]?.length ||
+                      !!baseStatusFilterTerm[cv.id]?.length) &&
+                    (patchStatusFilterTerm[cv.id]?.length
                       ? statuses.has(task.status)
                       : true) &&
-                    (baseStatusFilterTerm?.length
+                    (baseStatusFilterTerm[cv.id]?.length
                       ? baseStatuses.has(task.baseStatus)
                       : true),
                 }),
                 accumA
               ),
-            { ...selectedTasks }
+            { ...selectedTasks[cv.id] }
           ) ?? {};
```

Comment:
Woah this is pretty gnarly to read. I wonder if we can refactor this to make it easier to read.

Reply:
added to https://jira.mongodb.org/browse/EVG-15414